ls1intum/Artemis
# `Iris`: Load default templates from resource folder

### Checklist
#### General


#### Server
#### Client
No client changes

### Motivation and Context


The current solution for default Iris templates is to put them all into a single static utility class, `IrisConstants.java`. This is not scalable, offers no syntax highlighting, and forces the system to be recompiled every time a default template is changed.

### Description
Each default Iris template now gets its own `.hbs` (handlebars) file in the `resources/templates/iris` folder. The files are loaded using the `IrisDefaultTemplateService`, which uses the `ResourceLoaderService` to access the file with the requested filename in the folder. The `IrisSettingsService` then uses this service to load the default templates into the global settings at application startup.
The global template version used to be a static int field in IrisConstants, and this has also gotten its own file next to the templates. The content of this file is supposed to be just a single int and incremented every time one of the templates is adapted to trigger a reload from the files on application startup.

Please let me know if I have followed all best practices here!

### Steps for Testing

Prerequisites:
- 1 Administrator

1. Log in to Artemis
2. Navigate to Server Administration -> Global Iris Settings
3. Ensure that the contents of each template file have been loaded into their corresponding text areas in the settings, e.g. chat.hbs -> chat settings, hestia.hbs -> hestia settings, and the same for all code editor files.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)
